lib/service_actor/result.rb

```diff
@@ -51,5 +51,31 @@ def [](name)
     def display
       to_h.fetch(:display)
     end
+
+    def respond_to?(method_name, include_private = false)
+      !!method_missing(method_name) || super
+    end
+
+    private
+
+    def method_missing(symbol, *args)
+      attribute = symbol.to_s.chomp("?")
+
+      if symbol.end_with?("?") && respond_to?(attribute)
+        define_singleton_method symbol do
+          attribute_value = send(attribute.to_sym)
+
+          blank_proc = ->(value) { value.respond_to?(:empty?) ? !!value.empty? : !value }
+
+          # ActiveSupport ships with native #present?
+          # Delegate to ActiveSupport's Object#present? if ActiveSupport is present
+          attribute_value.respond_to?(:present?) ? attribute_value.present? : !blank_proc.call(attribute_value)
```

Comment:
Since we are in fact using the same code as ActiveSupport, what do you think about simplifying this down to the following?

```suggestion
          # Same as ActiveSupport’s #present?
          attribute_value.respond_to?(:empty?) ? !attribute_value.empty? : !!value
```

Reply:
I like your approach!. More cleaner and understandable!